python-frontend/src/main/java/org/sonar/python/semantic/SymbolTableBuilder.java

```diff
@@ -517,8 +521,53 @@ private void addSymbolUsage(Name nameTree) {
       SymbolImpl symbol = scope.resolve(nameTree.name());
       // TODO: use Set to improve performances
       if (symbol != null && symbol.usages().stream().noneMatch(usage -> usage.tree().equals(nameTree))) {
-        symbol.addUsage(nameTree, Usage.Kind.OTHER);
+          symbol.addUsage(nameTree, Usage.Kind.OTHER);
       }
     }
+
+    private void resolveParents(ClassDef classDef) {
+      Symbol symbol = classDef.name().symbol();
+      if (symbol == null || !Symbol.Kind.CLASS.equals(symbol.kind())) {
+        return;
+      }
+      ClassSymbolImpl classSymbol = (ClassSymbolImpl) symbol;
+      if (hasMultipleBindingUsages(classSymbol)) {
+        classSymbol.setKind(Symbol.Kind.OTHER);
+        return;
+      }
+      ArgList argList = classDef.args();
+      classSymbol.hasUnresolvedParents = false;
+      if (argList == null) {
+        return;
+      }
+      for (Argument argument : argList.arguments()) {
+        if (!argument.is(Kind.REGULAR_ARGUMENT) || !(((RegularArgument) argument).expression() instanceof HasSymbol)) {
+          classSymbol.hasUnresolvedParents = true;
+          return;
+        }
+        Symbol parentSymbol = ((HasSymbol) ((RegularArgument) argument).expression()).symbol();
+        if (parentSymbol == null) {
+          classSymbol.hasUnresolvedParents = true;
+          return;
+        }
+        if (hasMultipleBindingUsages(parentSymbol)) {
```

Comment:
I don't think we should have this check. We should add a symbol to parentSymbol even if it has multiple binding usages. It's then up to each rule to decide how to use those symbols.
Moreover setting `classSymbol.hasUnresolvedParents` is uncorrect here because we managed to find parent symbol (altough we don't know if it's a class or not.

Reply:
I removed the check as it was made redundant by the `Scope#addBindingUsage` change, but as discussed, the `classSymbol.hasUnresolvedParents` will still be set to true in that case (and the parent won't be added to the list).